codespell_lib/_codespell.py

```diff
@@ -862,7 +862,14 @@ def main(*args):
 
     file_opener = FileOpener(options.hard_encoding_detection,
                              options.quiet_level)
+
     glob_match = GlobMatch(options.skip)
+    try:
+        glob_match.match("/random/path")  # does not need a real path
+    except re.error:
+        print("ERROR: --skip/-S has been fed an invalid glob",
```

Comment:
I don't know if it's worth extending it to something like:
```suggestion
        print("ERROR: --skip/-S has been fed an invalid glob, try escaping special characters",
```

On the assumption a lot of people will just be doing `--skip foo.txt,bar.txt` etc and their filename just happens to have square brackets in it.